### DISCUSSION TITLE
How to initialize Airflow Connections in Docker

### DISCUSSION THREAD

**Discussion by gmyrianthous** [apache/airflow#30369] [Q&A]
Hi there, 

I have a `docker-compose.yml` file with the required services to run Airflow and a custom Dockerfile that I use to add custom dependencies etc. What I also want to do, is to be able to run `airflow connections` command in order to initialize the db with some pre-defined connections (and variables). 

My Dockerfile looks like this:

```
FROM apache/airflow:2.4.3-python3.8

COPY dags ./dags

COPY pyproject.toml .

# Install dependencies from pyproect.toml file
RUN pip install .

ENV PYTHONPATH "${PYTHONPATH}:/opt/airflow"

CMD airflow connections add 'my_prod_db' --conn-json '{"conn_type": "my-conn-type", "login": "my-login", "password": "my-password", "host": "my-host", "port": 1234, "schema": "my-schema", "extra": {"param1": "val1", "param2": "val2"}}'
```

Everything works fine except the part that creates the connection (which is the last `CMD` commnad in Dockerfile). I don't see any failures, but the connection is not created either. If I then connect to the worker `docker exec -it <airflow-worker-container-id> /bin/bash` and run the `airflow connections add ....` command, the connection is created just fine - however I would like to automate this step. 

Does anybody know what I am missing here? 

PS: I am aware that I can add connections by exporting environment variables, however I would like these connections to be visible on Airflow UI, and thus I want to avoid creating connections in the environment. 

Thanks in advance!

**Comment by potiuk** [comment]
yes. You cannot do that in docker image building because it operates on a database. You need to run this completely independently as separate step of deploying airflow (similarly as init command in the example of "quick-start" docker compose). You need to include that in whetever your process for airlfow installation is - and it is totally external from the docker compose itself. You do it only once after database initialization. 

BTW. You also have export/import options which might be better for you to use.

**Reply by gmyrianthous** [reply]
Thanks for your prompt reply. Can you please share a link to the init command of "quick-start" docker compose? 

Also, can you please elaborate on the export/import options?

**Reply by potiuk** [reply]
* https://airflow.apache.org/docs/apache-airflow/stable/howto/docker-compose/index.html
* Look at our docs and output of `airflow connections --help` command.

**Reply by gmyrianthous** [reply]
Isn't it possible to add the `airflow connections add` command as part of the `airflow-init` service?

**Reply by potiuk** [reply]
If you want - docker compose is up to you to modify as you want
